src/eko/quantities/__init__.py

```diff
@@ -0,0 +1,33 @@
+"""Physical and internal objects.
+
+This subpackage contains the definitions of many objects, with the purpose of
+using them as type hints, but not only (at least for some of them).
+
+Few related functions are also defined here.
+
+Note
+----
+Unfortunately, Python has still some discrepancies between runtime classes and
+type hints, so it is better not to mix dataclasses and generics.
+
+E.g. before it was implemented::
+
+    @dataclasses.dataclass
+    class RunningReference(DictLike, Generic[Quantity]):
+        value: Quantity
+        scale: float
+
+but in this way it is not possible to determine that ``RunningReference`` is
+subclassing ``DictLike``, indeed::
+
+    inspect.isclass(RunningReference)       # False
+    issubclass(RunningReference, DictLike)  # raise an error, since
+                                            # RunningReference is not a class
+
+Essentially classes can be used for type hints, but types are not all classes,
+especially when they involve generics.
+
+For this reason I prefer the less elegant dynamic generation, that seems to
```

Comment:
is this true for this subpackage?

Reply:
It was, it has been replaced by the `Generic` style, so I dropped dynamic generation.

However, I was aware that this comment is around (but good that you reminded me, because in the meanwhile I forgot), but I postponed the removal after confirmation that the new `Generic` based types were working as expected, with no further surprise.

Reply:
Done in 747956376a6503747da8edcd6a37da5265228b06 and 9fc14bafe7ee0e48ac2a043b2fd6926cddb6f018